src/compiler/emitter.ts

```diff
@@ -4716,6 +4719,9 @@ if (typeof __param !== "function") __param = function (paramIndex, decorator) {
                     write(unaliasedModuleNames.join(", "));
                 }
                 write("], function (require, exports");
+                if (compilerOptions.noEmitHelpers) {

```

Comment:
If the tslib runtime is going to handle these helpers is there anything from with say using: `tslib.extends` instead of `__extends`? That way we could do away with globals in all contexts/module patterns